File: app/src/main/java/org/dhis2/data/qr/QRjson.java
package org.dhis2.data.qr;

import android.util.Base64;

import java.io.UnsupportedEncodingException;

/**
 * QUADRAM. Created by ppajuelo on 22/05/2018.
 */

public class QRjson {

    public static final String EVENT_JSON = "EVENT";
    public static final String DATA_JSON_WO_REGISTRATION = "DATA_WO_REGISTRATION";
    public static final String DATA_JSON = "DATA";

    public static final String TEI_JSON = "TEI";
    public static final String ENROLLMENT_JSON = "ENROLLMENT";
    public static final String EVENTS_JSON = "EVENTS";
    public static final String ATTR_JSON = "ATTR";
    public static final String RELATIONSHIP_JSON = "RELATIONSHIP";

    private String type;
    private String data;

    QRjson(String type, String data) {
        this.type = type;
        this.data = data;
    }

    public String getType() {
        return type;
    }

    public String getData() throws UnsupportedEncodingException {
        String info;
        byte[] decodedBytes = Base64.decode(data, Base64.DEFAULT);
        info = new String(decodedBytes, "UTF-8");

        return info;
    }
}

File: app/src/main/java/org/dhis2/data/qr/QRjson.kt
package org.dhis2.data.qr

import android.util.Base64

import java.io.UnsupportedEncodingException

/**
 * QUADRAM. Created by ppajuelo on 22/05/2018.
 */

class QRjson internal constructor(val type: String, private val data: String) {

    @Throws(UnsupportedEncodingException::class)
    fun getData(): String {
        val info: String
        val decodedBytes = Base64.decode(data, Base64.DEFAULT)
        info = String(decodedBytes, Charsets.UTF_8)

        return info
    }

    companion object {
        const val EVENT_JSON = "EVENT"
        const val DATA_JSON_WO_REGISTRATION = "DATA_WO_REGISTRATION"
        const val DATA_JSON = "DATA"

        const val TEI_JSON = "TEI"
        const val ENROLLMENT_JSON = "ENROLLMENT"
        const val EVENTS_JSON = "EVENTS"
        const val ATTR_JSON = "ATTR"
        const val RELATIONSHIP_JSON = "RELATIONSHIP"
    }
}
